go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -452,9 +520,10 @@ type diffPartitions struct {
 	cmItr           doltdb.CommitItr
 	cmHashToTblInfo map[hash.Hash]tblInfoAtCommit
 	selectFunc      partitionSelectFunc
+	targetSch       schema.Schema
 }
 
-func newDiffPartitions(ctx *sql.Context, cmItr doltdb.CommitItr, wr *doltdb.RootValue, tblName string, selectFunc partitionSelectFunc) (*diffPartitions, error) {
+func newDiffPartitions(ctx *sql.Context, cmItr doltdb.CommitItr, wr *doltdb.RootValue, tblName string, selectFunc partitionSelectFunc, targetSch schema.Schema) (*diffPartitions, error) {
```

Comment:
Probably want schema param before iter param

Reply:
Best order here: ctx, tbl, sch, itr, root

The general principle (after context being first) is that when you are passing both schema and data, pass them in that order. If you have metadata (like table name), that goes first.

You might consider a little more of a refactor here. The only reason `wr` is passed is to get the underlying dolt table from it. So you could just pass that into this method in place of the table name and omit the root param entirely.